model_connection.go

```diff
@@ -64,7 +64,7 @@ func (c *ConnectionParameters) UnmarshalYAML(unmarshaller func(interface{}) erro
 	if err != nil {
 		return fmt.Errorf("failed to unserialize data (%w)", err)
 	}
-	c = &unserializedData
+	*c = unserializedData
```

Comment:
What does this do?

It looks to me like the old code threw away the unserialized data after writing its address to what I presume is a local parameter variable (`c`) which goes out of scope at the end of the function; now we are instead scribbling the unserialized data all over the `ConnectionParameters` object that `c` points to.  (I must be confused...why would Go permit that?!)

Where are we _supposed_ to put the unserialized data (address)?

Reply:
I am also confused by this line. I do not know what this is supposed to do. We should talk about this more in a meeting.

Reply:
So we're now replacing the `ConnectionParameters` receiver object rather than just overwriting the local receiver pointer.

The previous code didn't do anything, the new code does ... the question is whether that's correct. I suppose it depends on how and where this is called: but at least the new code does what the old code presumably wanted to do?

Reply:
After our discussion this morning, I no longer have an issue with this change:  the old code was apparently broken, and the new code seems entirely reasonable, especially if we think of this function as a constructor or initializer method.  And, given that, as far as we can tell, nothing actually uses this code, I think this change is low risk.  😆